core/store/src/trie/trie_storage.rs

```diff
@@ -133,31 +151,74 @@ impl TrieStorage for TrieMemoryPartialStorage {
 }
 
 /// Maximum number of cache entries.
-/// It was chosen to fit into RAM well. RAM spend on trie cache should not exceed
-/// 50_000 * 4 (number of shards) * TRIE_LIMIT_CACHED_VALUE_SIZE = 800 MB.
+/// It was chosen to fit into RAM well. RAM spend on trie cache should not exceed 50_000 * 4 (number of shards) *
+/// TRIE_LIMIT_CACHED_VALUE_SIZE * 2 (number of caches - for regular and view client) = 1.6 GB.
 /// In our tests on a single shard, it barely occupied 40 MB, which is dominated by state cache size
 /// with 512 MB limit. The total RAM usage for a single shard was 1 GB.
 #[cfg(not(feature = "no_cache"))]
-const TRIE_MAX_CACHE_SIZE: usize = 50000;
+const TRIE_MAX_SHARD_CACHE_SIZE: usize = 50000;
 
 #[cfg(feature = "no_cache")]
-const TRIE_MAX_CACHE_SIZE: usize = 1;
+const TRIE_MAX_SHARD_CACHE_SIZE: usize = 1;
 
 /// Values above this size (in bytes) are never cached.
 /// Note that Trie inner nodes are always smaller than this.
 const TRIE_LIMIT_CACHED_VALUE_SIZE: usize = 4000;
 
+/// Position of the value in cache.
+#[derive(Debug)]
+pub(crate) enum CachePosition {
```

Comment:
IMO CachePosition should refer to the position in the cache but this looks more like CacheType